### PR TITLE
Support casks in readall

### PR DESCRIPTION
**Pull request by SMillerDev**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

@MikeMcQuaid this should allow syntax checks for casks in CI

### CODE REVIEW

File: Library/Homebrew/readall.rb [hunk 1]
```diff
@@ -48,14 +49,29 @@ def valid_formulae?(formulae)
...
+      casks.each do |file|
+        Cask::CaskLoader.load(file)
+      rescue Interrupt
+        raise
+      rescue Exception => e # rubocop:disable Lint/RescueException
+        onoe "Invalid cask: #{file}"
+        puts e
+        failed = true
+      end
+      !failed
```
Comment (reitermarkus) [inline Library/Homebrew/readall.rb:63]: Maybe name it `success` instead so we don't need the negation here?

File: Library/Homebrew/readall.rb [hunk 2]
```diff
@@ -48,14 +49,29 @@ def valid_formulae?(formulae)
...
 
+    def valid_casks?(casks)
+      failed = false
+      casks.each do |file|
+        Cask::CaskLoader.load(file)
+      rescue Interrupt
+        raise
+      rescue Exception => e # rubocop:disable Lint/RescueException
+        onoe "Invalid cask: #{file}"
+        puts e
```
Comment (reitermarkus) [inline Library/Homebrew/readall.rb:60]: `$stderr.puts`

File: Library/Homebrew/readall.rb (hunk 1 above)
Comment (SMillerDev) [reply Library/Homebrew/readall.rb:63]: I'll perform that change everywhere then

File: Library/Homebrew/readall.rb [hunk 3]
```diff
@@ -48,14 +49,29 @@ def valid_formulae?(formulae)
...
     def valid_tap?(tap, options = {})
       failed = false
       if options[:aliases]
         valid_aliases = valid_aliases?(tap.alias_dir, tap.formula_dir)
         failed = true unless valid_aliases
       end
       valid_formulae = valid_formulae?(tap.formula_files)
-      failed = true unless valid_formulae
+      valid_casks = valid_casks?(tap.cask_files)
+      failed = true unless valid_formulae || valid_casks
```
Comment (MikeMcQuaid) [inline Library/Homebrew/readall.rb:74]:
```suggestion
      failed = true if !valid_formulae || !valid_casks
```

**Comment by SMillerDev** [comment]
@MikeMcQuaid CI says no

**Comment by MikeMcQuaid** [comment]
@SMillerDev `brew man` wants run

File: Library/Homebrew/readall.rb [hunk 4]
```diff
@@ -35,28 +36,43 @@ def valid_aliases?(alias_dir, formula_dir)
...
+      success
     end
 
     def valid_tap?(tap, options = {})
-      failed = false
+      success = true
       if options[:aliases]
         valid_aliases = valid_aliases?(tap.alias_dir, tap.formula_dir)
-        failed = true unless valid_aliases
+        success = false if !valid_aliases
```
Comment (MikeMcQuaid) [inline Library/Homebrew/readall.rb:70]: Sorry, yeh, this should be `unless`

**Comment by SMillerDev** [comment]
I don't think the failure has anything to do with my code?

**Comment by MikeMcQuaid** [comment]
> Cannot tap homebrew/cask: invalid syntax in tap!

That's definitely related, I'm afraid.

**Comment by SMillerDev** [comment]
I can't reproduce this locally.

**Comment by MikeMcQuaid** [comment]
@SMillerDev on Linux? That's where it's failing. It seems unsurprising that reading taps would fail there 😄

**Comment by SMillerDev** [comment]
*grabs more coffee* 

I guess I need to return casks as successful on Linux then?

File: Library/Homebrew/readall.rb [hunk 5]
```diff
@@ -35,28 +36,43 @@ def valid_aliases?(alias_dir, formula_dir)
...
-        puts e
-        failed = true
+        $stderr.puts e
+        success = false
       end
-      !failed
+      success
+    end
+
+    def valid_casks?(casks)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/readall.rb:52]: Could add a `os/linux` implementation that makes this return `true` unconditionally

File: Library/Homebrew/readall.rb (hunk 4 above)
Comment (reitermarkus) [reply Library/Homebrew/readall.rb]: Maybe use `success &= valid_aliases` for all of these?

**Comment by SMillerDev** [comment]
I can't manage to figure this out, closing.

**Comment by MikeMcQuaid** [comment]
@SMillerDev I'll get this finished off for you.

**Comment by SMillerDev** [comment]
🥳

**Pull request merged**